app/scripts/controllers/app-state-controller.test.ts

```diff
@@ -43,680 +46,648 @@ const extensionMock = {
-      expect(updateStateSpy).toHaveBeenCalledWith({
-        termsOfUseLastAgreed: mockParams,
+        expect(controller.state.termsOfUseLastAgreed).toStrictEqual(mockParams);
       });
-      expect(
-        appStateController.store.getState().termsOfUseLastAgreed,
-      ).toStrictEqual(mockParams);
-
-      updateStateSpy.mockRestore();
     });
   });
 
   describe('onPreferencesStateChange', () => {
-    it('should update the timeoutMinutes with the autoLockTimeLimit', () => {
-      ({ appStateController, controllerMessenger } =
-        createAppStateController());
-      const timeout = Date.now();
-
-      controllerMessenger.publish(
-        'PreferencesController:stateChange',
-        {
-          preferences: { autoLockTimeLimit: timeout },
-        } as unknown as PreferencesControllerState,
-        [],
-      );
-
-      expect(appStateController.store.getState().timeoutMinutes).toStrictEqual(
-        timeout,
-      );
+    it('should update the timeoutMinutes with the autoLockTimeLimit', async () => {
+      await withController(({ controller, controllerMessenger }) => {
+        const timeout = Date.now();
+
+        controllerMessenger.publish(
+          'PreferencesController:stateChange',
+          {
+            preferences: { autoLockTimeLimit: timeout },
+          } as unknown as PreferencesControllerState,
+          [],
+        );
+
+        expect(controller.state.timeoutMinutes).toStrictEqual(timeout);
+      });
     });
   });
 
   describe('isManifestV3', () => {
-    it('creates alarm when isManifestV3 is true', () => {
+    it('creates alarm when isManifestV3 is true', async () => {
       mockIsManifestV3.mockReturnValue(true);
-      ({ appStateController } = createAppStateController());
-
-      const timeout = Date.now();
-      controllerMessenger.publish(
-        'PreferencesController:stateChange',
-        {
-          preferences: { autoLockTimeLimit: timeout },
-        } as unknown as PreferencesControllerState,
-        [],
-      );
-      const spy = jest.spyOn(
-        appStateController as unknown as { _resetTimer: () => void },
-        '_resetTimer',
-      );
-      appStateController.setLastActiveTime();
-
-      expect(spy).toHaveBeenCalled();
-      expect(extensionMock.alarms.clear).toHaveBeenCalled();
-      expect(extensionMock.alarms.onAlarm.addListener).toHaveBeenCalled();
+      await withController(({ controller, controllerMessenger }) => {
+        const timeout = Date.now();
+        controllerMessenger.publish(
+          'PreferencesController:stateChange',
+          {
+            preferences: { autoLockTimeLimit: timeout },
+          } as unknown as PreferencesControllerState,
+          [],
+        );
+        controller.setLastActiveTime();
+
+        expect(extensionMock.alarms.clear).toHaveBeenCalled();
+        expect(extensionMock.alarms.onAlarm.addListener).toHaveBeenCalled();
+      });
     });
   });
 
   describe('AppStateController:getState', () => {
```

Comment:
What are your thoughts on removing the tests for the `getState` event? This is something that's already tested at the BaseController layer.